src/DotVVM.Framework/Routing/RouteHelper.cs

```diff
@@ -35,5 +36,54 @@ public static void AutoDiscoverRoutes(this DotvvmRouteTable table, IRoutingStrat
                 table.Add(route.RouteName, route);
             }
         }
+
+        public static void AssertConfigurationIsValid(this DotvvmConfiguration config)
+        {
+            var invalidRoutes = new List<DotvvmConfigurationAssertResult<RouteBase>>();
+            var invalidControls = new List<DotvvmConfigurationAssertResult<DotvvmControlConfiguration>>();
+            var loader = new AggregateMarkupFileLoader();
+            foreach (var route in config.RouteTable.Where(s => !string.IsNullOrWhiteSpace(s.VirtualPath)))
+            {
+                if (string.IsNullOrWhiteSpace(route.RouteName))
+                {
+                    invalidRoutes.Add(new DotvvmConfigurationAssertResult<RouteBase>(route, DotvvmConfigurationAssertReason.MissingRouteName));
+                }
+
+                var content = loader.GetMarkup(config, route.VirtualPath);
```

Comment:
This check is invalid - you can register you own loader that will understand the path and load it correctly.

Reply:
We can provide optional parameter IMarkupFileLoader.

Reply:
I don't see any, where is the parameter? You are initializing the `loader` here: https://github.com/riganti/dotvvm/pull/578/files/a4ffcbd72298c4a8d8d71626ff6627f4cc97c280#diff-c9adfdb58f0b70c3f469077f400c8d38R44

And why make things complicated, you can simply get the loader from service collection.

Reply:
I don't think we need an optional parameter. The loader is retrieved from `ServiceProvider` - it's good enough. But I agree, this check is invalid.

Reply:
Or maybe not. The loader does not work with the route itself. And the `VirtualPath` is not validated here.

Reply:
Originally the loader was not retrieved from the `ServiceProvider`, my comment was about that. But you are right, that the virtual path is actually interpreted by the `IDotvvmPresenter`, so the check is invalid anyway...